artemis-commons/src/main/java/org/apache/activemq/artemis/core/server/ActiveMQScheduledComponent.java

```diff
@@ -265,6 +270,9 @@ public void stop() {
       if (future != null) {
          future.cancel(false);
          future = null;
+         // Replace te existing one: a new periodic task or any new delay after stop
+         // won't interact with the previously running ones
+         this.bookedForRunning = new AtomicBoolean(false);
```

Comment:
The comment doesnt entirely fit if this is inside the null check, does it? It wont apply for a 'new delay' since the future doesnt exist for such 'on demand' usage.